### PR TITLE
add ignore list notes by modifying FriendNotes plugin

### PR DESCRIPTION
**Pull request by Rami-J**
Closes Issue: #10442 

Adds notes for players on the ignore list, by modifying the FriendNotes plugin.

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/friendnotes/FriendNotesPlugin.java [hunk 1]
```diff
@@ -61,10 +62,13 @@
 public class FriendNotesPlugin extends Plugin
 {
 	private static final String CONFIG_GROUP = "friendNotes";
+	private static final String CONFIG_GROUP_IGNORE = "ignoreNotes";
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/friendnotes/FriendNotesPlugin.java]: It would probably make more sense to reuse the config group for friendnotes, since a player cannot be on both your friend list and ignore list, and it would let the note migrate between lists.

File: runelite-client/src/main/java/net/runelite/client/plugins/hiscore/HiscorePlugin.java [hunk 2]
```diff
@@ -177,11 +177,12 @@ public void onMenuEntryAdded(MenuEntryAdded event)
...
 		if (groupId == WidgetInfo.FRIENDS_LIST.getGroupId() || groupId == WidgetInfo.CLAN_CHAT.getGroupId() ||
 				groupId == WidgetInfo.CHATBOX.getGroupId() && !KICK_OPTION.equals(option) || //prevent from adding for Kick option (interferes with the raiding party one)
-				groupId == WidgetInfo.RAIDING_PARTY.getGroupId() || groupId == WidgetInfo.PRIVATE_CHAT_MESSAGE.getGroupId())
+				groupId == WidgetInfo.RAIDING_PARTY.getGroupId() || groupId == WidgetInfo.PRIVATE_CHAT_MESSAGE.getGroupId() ||
+				groupId == WidgetInfo.IGNORE_LIST.getGroupId())
 		{
 			boolean after;
 
-			if (!AFTER_OPTIONS.contains(option))
+			if (!AFTER_OPTIONS.contains(option) || (option.equals("Delete") && groupId != WidgetInfo.IGNORE_LIST.getGroupId()))
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/hiscore/HiscorePlugin.java]: Is this for adding lookup to ignore list? This should probably be in a separate PR

Comment (Rami-J) [reply runelite-client/src/main/java/net/runelite/client/plugins/hiscore/HiscorePlugin.java]: Yes, I made a separate PR for it now

> Is this for adding lookup to ignore list? This should probably be in a separate PR

**Comment by Adam-** [comment]
The approach this is using to detect when a player is unignored won't work if you eg. type in a player name to remove. I am unsure if removing the note is useful when unignored though, why is it being done?

**Comment by Rami-J** [comment]
Oh, that is something I didn't realize. I can remove that check then. I wanted to mirror the behavior of removing a friend with a note.

**Comment by Adam-** [comment]
Okay. I modified the event now for ignores too so this should just work: 608ec68ae65978988dc12301ece6d37ae7e33907
